com.unity.perception/Runtime/GroundTruth/Labelers/BoundingBox3DLabeler.cs

```diff
@@ -0,0 +1,277 @@
+                instance_id = instanceId,
+                translation = new[] { center.x, center.y, center.z },
+                size = new[] { extents.x * 2, extents.y * 2, extents.z * 2 },
+                rotation = new[] { rotation.x, rotation.y, rotation.z, rotation.w },
+                velocity = null,
+                acceleration = null
+            };
+        }
+
+        static Vector3[] GetBoxCorners(Bounds bounds, Quaternion rotation)
+        {
+            var boundsCenter = bounds.center;
+            var right = Vector3.right * bounds.extents.x;
+            var up = Vector3.up * bounds.extents.y;
+            var forward = Vector3.forward * bounds.extents.z;
+
+            right = rotation * right;
+            up = rotation * up;
+            forward = rotation * forward;
+            
+            var doubleRight = right * 2;
+            var doubleUp = up * 2;
+            var doubleForward = forward * 2;
+            
+            var corners = new Vector3[8];
+            corners[0] = boundsCenter - right - up - forward;
+            corners[1] = corners[0] + doubleUp;
+            corners[2] = corners[1] + doubleRight;
+            corners[3] = corners[0] + doubleRight; 
+            for (var i = 0; i < 4; i++)
+            {
+                corners[i + 4] = corners[i] + doubleForward;
+            }
+
+            return corners;
+        }
+
+        /// <inheritdoc/>
+        protected override void OnBeginRendering()
+        {
+            if (m_BoundingBoxValues == null)
+                m_BoundingBoxValues = new List<BoxData>();
+            else
+                m_BoundingBoxValues.Clear();
+
+            m_PerceptionCameraFrustumFrames = GeometryUtility.CalculateFrustumPlanes(perceptionCamera.attachedCamera);
+            m_CurrentFrame = Time.frameCount;
+
+            var entities = m_EntityQuery.ToEntityArray(Allocator.TempJob);
+            var entityManager = World.DefaultGameObjectInjectionWorld.EntityManager;
+            
+            foreach (var entity in entities)
+            {
+                ProcessEntity(entityManager.GetComponentObject<Labeling>(entity));
+            }
+
+            entities.Dispose();
+            
+            perceptionCamera.SensorHandle.ReportAnnotationAsync(m_AnnotationDefinition).ReportValues(m_BoundingBoxValues);
+        }
+
+        bool IsInCameraView(GameObject target)
+        {
+            // Need to check against all of the meshes in the labeled object, if at least one is visible, then
+            // the object will be considered in the camera's view. The built in Unity geometry methods are optimized to
+            // work with AABB bounds, so we will use those from the mesh renderer, as opposed to the non-axis aligned
+            // bounding boxes retrieved from the mesh filter. There is a chance that this method can report false
+            // positives, but it is (at least right now) probably worth the trade off for performance
```

Comment:
There appears to be a fairly high chance of false positives here, especially when objects are occluded. It's a bit more work, but this would be addressed if we use the RenderedObjectInfo to filter the set of bounding boxes before reporting them.

Reply:
fixed.